lnwallet/channel.go

```diff
@@ -1893,6 +1893,33 @@ func (lc *LightningChannel) restorePendingRemoteUpdates(
 				payDesc.LogIndex)
 		}
 
+		var height uint64
+		var heightSet bool
+		if payDesc.EntryType != Add {
+			// Check whether this update is included on the pending commitment
+			// If so, we should set the removeCommitHeightRemote.
+			pendingRemoteCommitDiff, err := lc.channelState.RemoteCommitChainTip()
+			if err != nil && err != channeldb.ErrNoPendingCommit {
+				return err
+			}
+
+			if pendingRemoteCommitDiff != nil {
+				pendingCommitPoint := lc.channelState.RemoteNextRevocation
+				pendingRemoteCommit, err := lc.diskCommitToMemCommit(
+					false, &pendingRemoteCommitDiff.Commitment,
+					nil, pendingCommitPoint,
+				)
+				if err != nil {
+					return err
+				}
+
+				if payDesc.LogIndex < pendingRemoteCommit.theirMessageIndex {
```

Comment:
Yeah seems all we need here is the message index of the pending remote commit, so we can avoid having to look it up again.